gobblin-data-management/src/main/java/gobblin/data/management/retention/policy/WhiteListPredicateRetentionPolicy.java

```diff
@@ -0,0 +1,52 @@
+/*
+* Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+* this file except in compliance with the License. You may obtain a copy of the
+* License at  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed
+* under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, either express or implied.
+*/
+
+package gobblin.data.management.retention.policy;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+import gobblin.data.management.retention.version.DatasetVersion;
+
+
+public class WhiteListPredicateRetentionPolicy implements RetentionPolicy<DatasetVersion> {
+
+  private final Predicate<DatasetVersion> whitelistPredicate;
+
+  private static final String WHITELIST_RETENTION_POLICY_PREDICATE_NAME =
+      "gobblin.retention.whitelist.retention.policy.predicate";
+
+  @SuppressWarnings("unchecked")
+  public WhiteListPredicateRetentionPolicy(Properties props) throws InstantiationException, IllegalAccessException,
+      ClassNotFoundException, IllegalArgumentException, SecurityException, InvocationTargetException,
+      NoSuchMethodException {
+    this.whitelistPredicate =
+        (Predicate<DatasetVersion>) Class.forName(props.getProperty(WHITELIST_RETENTION_POLICY_PREDICATE_NAME))
+            .getConstructor(Properties.class).newInstance(props);
+  }
+
+  @Override
+  public Class<? extends DatasetVersion> versionClass() {
+    return DatasetVersion.class;
+  }
+
+  @Override
+  public Collection<DatasetVersion> listDeletableVersions(List<DatasetVersion> allVersions) {
+    return Lists.newArrayList(Iterables.filter(allVersions, this.whitelistPredicate));
```

Comment:
This will delete the versions that pass the predicate. IMO it is more natural to delete the versions that fail the predicate. Any thoughts?
This should be well documented either way.